### PR TITLE
ibmcloud: switch to k8s Community-Owned Package Repositories

### LINKED ISSUE #1729: ibmcloud: Ansible task failed: Add kubernetes source list
Got errors as below when create ibm cloud using ansible
```
TASK [Add gpg key]
*************************************************************
changed: [162.133.136.104]
changed: [162.133.128.92]

TASK [Add kubernetes source list]
**********************************************
fatal: [162.133.136.104]: FAILED! => {"changed": false, "msg": "Failed to
update apt cache: E:The repository 'http://packages.cloud.google.com/apt
kubernetes-xenial Release' does not have a Release file."}
fatal: [162.133.128.92]: FAILED! => {"changed": false, "msg": "Failed to
update apt cache: E:The repository 'http://packages.cloud.google.com/apt
kubernetes-xenial Release' does not have a Release file."}
```
This is reported at https://github.com/confidential-containers/cloud-api-adaptor/blob/main/ibmcloud/cluster/ansible/tasks/k8s.yaml#L14,

And can be reproduced easily on a ubuntu linux by 
```
root:~$ curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo gpg --dearmour -o /etc/apt/trusted.gpg.d/kubernetes-xenial.gpg
sudo apt-add-repository "deb http://apt.kubernetes.io/ kubernetes-xenial main"
Repository: 'deb http://apt.kubernetes.io/ kubernetes-xenial main'
Description:
Archive for codename: kubernetes-xenial components: main
More info: http://apt.kubernetes.io/
Adding repository.
Press [ENTER] to continue or Ctrl-c to cancel.
Adding deb entry to /etc/apt/sources.list.d/archive_uri-http_apt_kubernetes_io_-jammy.list
Adding disabled deb-src entry to /etc/apt/sources.list.d/archive_uri-http_apt_kubernetes_io_-jammy.list
Hit:1 http://us.archive.ubuntu.com/ubuntu jammy InRelease
Get:2 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Hit:3 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease
Get:4 http://us.archive.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Ign:5 https://packages.cloud.google.com/apt kubernetes-xenial InRelease
Err:6 https://packages.cloud.google.com/apt kubernetes-xenial Release
  404  Not Found [IP: 172.253.115.100 443]
Get:7 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [1433 kB]
Get:8 http://us.archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [1523 kB]
Get:9 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1053 kB]
Get:10 http://us.archive.ubuntu.com/ubuntu jammy-security/main amd64 Packages [1217 kB]
Get:11 http://us.archive.ubuntu.com/ubuntu jammy-security/main Translation-en [220 kB]
Get:12 http://us.archive.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [1495 kB]
Get:13 http://us.archive.ubuntu.com/ubuntu jammy-security/restricted Translation-en [248 kB]
Reading package lists... Done
E: The repository 'http://apt.kubernetes.io kubernetes-xenial Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

### PR DESCRIPTION
**Pull request by lysliu**
fixes #1729 

switch to k8s Community-Owned Package Repositories

### CODE REVIEW

**Review by liudalibj — APPROVED** [review]
LGTM
Thanks @lysliu !

File: ibmcloud/cluster/ansible/tasks/k8s.yaml [hunk 1]
```diff
@@ -6,12 +6,12 @@
 ---
 - name: "Add gpg key "
   apt_key:
-    url: https://packages.cloud.google.com/apt/doc/apt-key.gpg
+    url: https://pkgs.k8s.io/core:/stable:/v1.28/deb/Release.key
```
Comment (huoqifeng) [inline ibmcloud/cluster/ansible/tasks/k8s.yaml:9]: Sounds like the k8s version is same as kubeadm by default as described in https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#kubeadm-s-skew-against-the-kubernetes-version, I'm wondering is it possible to get the version from kubadm --version and set it dynamically?

**Review by huoqifeng — APPROVED** [review]
LGTM, I saw the log successfully after the change thanks! @lysliu

**Comment by huoqifeng** [comment]
I also triggered the CI process.

**Comment by huoqifeng** [comment]
govulncheck failure is as expected, all others passed, I'm going to merge it.

**Pull request merged**